pkg/virt-launcher/virtwrap/agent-poller/agent_poller_test.go

```diff
@@ -29,15 +29,71 @@ import (
 			json_input := "{\"return\":[{\"name\":\"lo\",\"ip-addresses\":[{\"ip-address-type\":\"ipv4\",\"ip-address\":\"127.0.0.1\",\"prefix\":8},{\"ip-address-type\":\"ipv6\",\"ip-address\":\"::1\",\"prefix\":128}],\"hardware-address\":\"00:00:00:00:00:00\"},{\"name\":\"eth0\",\"ip-addresses\":[{\"ip-address-type\":\"ipv4\",\"ip-address\":\"10.244.0.81\",\"prefix\":24},{\"ip-address-type\":\"ipv6\",\"ip-address\":\"fe80::858:aff:fef4:51\",\"prefix\":64}],\"hardware-address\":\"0a:58:0a:f4:00:51\"},{\"name\":\"eth1\",\"ip-addresses\":[{\"ip-address-type\":\"ipv6\",\"ip-address\":\"fe80::ff:feb0:1766\",\"prefix\":64}],\"hardware-address\":\"02:00:00:b0:17:66\"}, {\"name\":\"eth5\",\"ip-addresses\":[{\"ip-address-type\":\"ipv4\",\"ip-address\":\"1.2.3.4\",\"prefix\":24},{\"ip-address-type\":\"ipv6\",\"ip-address\":\"fe80::ff:1111:2222\",\"prefix\":64}],\"hardware-address\":\"02:00:00:22:11:11\"}]}"
-			interfaceStatuses := agentPoller.GetInterfaceStatuses(json_input)
+			interfaceStatuses := parseInterfaces(json_input)
+			expectedStatuses := []api.InterfaceStatus{}
+			expectedStatuses = append(expectedStatuses,
+				api.InterfaceStatus{
+					Name:          "",
+					Mac:           "0a:58:0a:f4:00:51",
+					Ip:            "10.244.0.81/24",
+					IPs:           []string{"10.244.0.81/24", "fe80::858:aff:fef4:51/64"},
+					InterfaceName: "eth0",
+				})
+			expectedStatuses = append(expectedStatuses,
+				api.InterfaceStatus{
+					Name:          "",
+					Mac:           "02:00:00:b0:17:66",
+					Ip:            "fe80::ff:feb0:1766/64",
+					IPs:           []string{"fe80::ff:feb0:1766/64"},
+					InterfaceName: "eth1",
+				})
+			expectedStatuses = append(expectedStatuses,
+				api.InterfaceStatus{
+					Mac:           "02:00:00:22:11:11",
+					Ip:            "1.2.3.4/24",
+					IPs:           []string{"1.2.3.4/24", "fe80::ff:1111:2222/64"},
+					InterfaceName: "eth5",
+				})
+			Expect(interfaceStatuses).To(Equal(expectedStatuses))
+		})
+
+		It("should merge QEMU info and agent info", func() {
+			// agentPoller := AgentPoller{}
+
+			json_input := "{\"return\":[{\"name\":\"lo\",\"ip-addresses\":[{\"ip-address-type\":\"ipv4\",\"ip-address\":\"127.0.0.1\",\"prefix\":8},{\"ip-address-type\":\"ipv6\",\"ip-address\":\"::1\",\"prefix\":128}],\"hardware-address\":\"00:00:00:00:00:00\"},{\"name\":\"eth0\",\"ip-addresses\":[{\"ip-address-type\":\"ipv4\",\"ip-address\":\"10.244.0.81\",\"prefix\":24},{\"ip-address-type\":\"ipv6\",\"ip-address\":\"fe80::858:aff:fef4:51\",\"prefix\":64}],\"hardware-address\":\"0a:58:0a:f4:00:51\"},{\"name\":\"eth1\",\"ip-addresses\":[{\"ip-address-type\":\"ipv6\",\"ip-address\":\"fe80::ff:feb0:1766\",\"prefix\":64}],\"hardware-address\":\"02:00:00:b0:17:66\"}, {\"name\":\"eth5\",\"ip-addresses\":[{\"ip-address-type\":\"ipv4\",\"ip-address\":\"1.2.3.4\",\"prefix\":24},{\"ip-address-type\":\"ipv6\",\"ip-address\":\"fe80::ff:1111:2222\",\"prefix\":64}],\"hardware-address\":\"02:00:00:22:11:11\"}]}"
```

Comment:
This is really hard to read. Being able to read this as plain multiline json without lots of escaped double quotes would really help me to have a more clear view of what's going on here. Maybe you could convert the test data into a multi line string?

Reply:
You are right, fixed that in the whole test file. Thanks for pointing it up, I was so used to it, that it did not occurred to me.